code/game/objects/items/weapons/storage/quickdraw_vr.dm

```diff
@@ -0,0 +1,79 @@
+// -----------------------------
+//       Quickdraw storage
+// -----------------------------
+//These items are pouches and cases made to be kept in belts or pockets to quickly draw objects from
+//Largely inspired by the vest pouches on Colonial Marines
+
+/obj/item/weapon/storage/quickdraw
+	name = "quickdraw"
+	desc = "This object should not appear"
+	icon = 'icons/obj/storage_vr.dmi'
+
+	//Quickmode
+	//When set to 0, this storage will operate as a regular storage, and clicking on it while equipped will open it as a storage
+	//When set to 1, a click while it is equipped will instead move the first item inside it to your hand
+	var/quickmode = 0
+
+/obj/item/weapon/storage/quickdraw/attack_hand(mob/user as mob)
+	if(src.loc == user) //If they aren't holding us, we do nothing special
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			if(quickmode)
+				var/first_item = contents[1]
+				if(first_item && !H.get_active_hand()) //Do we have anything to give you?
+					H.put_in_hands(first_item)
+					contents[1] = null
+					return
+
+			if(H.l_store == src && !H.get_active_hand()) //overrides
+				src.open(user)
+				return
+			if(H.r_store == src && !H.get_active_hand())
+				src.open(user)
+				return
+	..() //Nothing special happened, go call the other proc
+
+
+/obj/item/weapon/storage/quickdraw/verb/toggle_quickdraw()
+	set name = "Switch Quickdraw Mode"
+	set category = "Object"
+
+	quickmode = !quickmode
+	switch (quickmode)
+		if(1)
+			to_chat(usr, "[src] now draws the first object inside.")
+		if(0)
+			to_chat(usr, "[src] now opens as a container.")
+
+/obj/item/weapon/storage/quickdraw/AltClick(mob/user)
+	..()
+	if(src.loc == user) //They holding us?

```

Comment:
Not really important, but comment misleading. This would work from a pocket or any other inventory slot. Comment implies being held.

Reply:
True. I'll amend it I suppose